xclim/core/bootstrapping.py

```diff
@@ -0,0 +1,191 @@
+from inspect import signature
+from typing import Any, Callable, Dict
+
+import numpy as np
+import xarray as xr
+from boltons.funcutils import wraps
+from xarray.core.dataarray import DataArray
+
+from xclim.core.calendar import convert_calendar, parse_offset, percentile_doy
+
+
+def percentile_bootstrap(func):
+    """Decorator applying a bootstrap step to the calculation of exceedance over a percentile threshold.
+
+    Boostraping avoids discontinuities in the exceedance between the "in base" period over which percentiles are
+    computed, and "out of base" periods. See `bootstrap_func` for details.
+
+    Example of declaration:
+    @declare_units(tas="[temperature]", t90="[temperature]")
+    @percentile_bootstrap
+    def tg90p(
+        tas: xarray.DataArray,
+        t90: xarray.DataArray,
+        freq: str = "YS",
+        bootstrap: bool = False
+    ) -> xarray.DataArray:
+
+    Example when called:
+    >>> import xarray as xr
```

Comment:
```suggestion
```
The magic going on here is that the `xr` used in the doctests is not `xarray`, but actually points to `xclim.testing._utils`:
https://github.com/Ouranosinc/xclim/blob/675bee51a51e29298d349578d8532e35f9686457/xclim/testing/_utils.py#L82
This function lets us set up examples using a custom `open_dataset` that looks for our test datasets on your disk or pulls it from our testdata repo. Removing the import will fix this. Sorry for the confusion!